src/domain/event/CreateEventPage.tsx

```diff
@@ -23,11 +32,46 @@ const CreateEventPage: React.FC = () => {
 
   const [createEvent] = useCreateEventMutation();
   const [updateImage] = useUpdateSingleImageMutation();
+  const [updateVenue] = useEditVenueMutation();
+  const [createVenue] = useCreateVenueMutation();
 
   const submit = async (values: EventFormFields) => {
     try {
       const requests = [];
 
+      const venueData = apolloClient.readQuery<VenueQuery>({
+        query: VenueDocument,
+        variables: { id: values.location },
+      });
+
+      const venueShouldBeUpdated = Boolean(
+        venueData?.venue && values.locationDescription
```

Comment:
You could compare values.locationDescription to venue description (helper function to get that). Now it's not possible to clear venue description
```
const venueDescription =
    eventData?.event?.venue?.translations.find(
      (t) => t.languageCode === language.toUpperCase()
    )?.description || '';

const venueShouldBeUpdated = Boolean(
        venueData?.venue && values.locationDescription !== venueDescription
```